nifi-extension-bundles/nifi-network-bundle/nifi-network-processors/src/main/java/org/apache/nifi/processors/network/pcap/SplitPCAP.java

```diff
@@ -0,0 +1,245 @@
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return DESCRIPTORS;
+    }
+
+    /**
+     * This method is called when a trigger event occurs in the processor.
+     * It processes the incoming flow file, splits it into smaller pcap files based on the PCAP Max Size,
+     * and transfers the split pcap files to the success relationship.
+     * If the flow file is empty or not parseable, it is transferred to the failure relationship.
+     *
+     * @param context  the process context
+     * @param session  the process session
+     */
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) {
+
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final List<FlowFile> splitFiles = new ArrayList<>();
+
+        try {
+            session.read(flowFile, new InputStreamCallback() {
+                @Override
+                public void process(final InputStream inStream) {
+                    try {
+                        final int pcapMaxSize = context.getProperty(PCAP_MAX_SIZE.getName()).asDataSize(DataUnit.B).intValue();
+                        final List<Packet> loadedPackets = new ArrayList<>();
+                        final BufferedInputStream bInStream = new BufferedInputStream(inStream);
+
+                        if ( bInStream.available() == 0 ) {
+                            throw new IOException("PCAP file empty.");
+                        }
+
+                        final byte[] pcapHeader = new byte[PCAP.PCAP_HEADER_LENGTH];
+                        bInStream.read(pcapHeader, 0, PCAP.PCAP_HEADER_LENGTH);
+                        int currentPcapTotalLength = PCAP.PCAP_HEADER_LENGTH;
+
+                        final PCAP templatePcap = new PCAP(new ByteBufferInterface(pcapHeader));
+
+                        while (bInStream.available() > 0) {
+
+                            byte[] packetHeader = new byte[Packet.PACKET_HEADER_LENGTH];
+                            bInStream.read(packetHeader, 0, Packet.PACKET_HEADER_LENGTH);
+                            Packet currentPacket = new Packet(packetHeader, templatePcap);
+
+                            if (currentPacket.totalLength() > pcapMaxSize) {
+                                throw new IOException("PCAP contains a packet larger than the max size.");
+                            }
+
+                            if (currentPacket.isInvalid()) {
+                                throw new IOException("PCAP contains an invalid packet.");
+                            }
+
+                            byte[] packetBodies = new byte[(int) currentPacket.expectedLength()];
+
+                            bInStream.read(packetBodies, 0, (int) currentPacket.expectedLength());
+                            currentPacket.setBody(packetBodies);
```

Comment:
@JackHintonSmartDCSIT Based on line 178 where the method is named `setBody` shouldn't the variable on line 175 be `packetBody`? My earlier comment about plural is for a collection of items which represent uniquely different items while in this case the array here represents a single packet body.

Reply:
Damnit, I knew I forgot one. I'll fix that now